packages/serverpod_shared/test/config_test.dart

```diff
@@ -0,0 +1,329 @@
+      () {
+    var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+database:
+  host: localhost
+  port: 5432
+  name: testDb
+  user: test
+''';
+
+    expect(
+      () => ServerpodConfig.loadFromYaml(
+        '',
+        '',
+        {},
+        loadYaml(serverpodConfig),
+      ),
+      throwsA(isA<Exception>().having(
+        (e) => e.toString(),
+        'message',
+        equals('Exception: Missing database password.'),
+      )),
+    );
+  });
+
+  test(
+      'Given Serverpod config with database configuration missing required field when loading from Yaml then exception is thrown.',
+      () {
+    var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+database:
+  host: localhost
+  name: testDb
+  user: test
+''';
+
+    expect(
+      () => ServerpodConfig.loadFromYaml(
+        '',
+        '',
+        {'database': 'password'},
+        loadYaml(serverpodConfig),
+      ),
+      throwsA(isA<Exception>().having(
+        (e) => e.toString(),
+        'message',
+        equals(
+            'Exception: database is missing required configuration for port.'),
+      )),
+    );
+  });
+
+  test(
+      'Given Serverpod config with database configuration when loading from Yaml then config is loaded correctly.',
+      () {
+    var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+database:
+  host: localhost
+  port: 5432
+  name: testDb
+  user: test
+''';
+
+    var config = ServerpodConfig.loadFromYaml(
+      '',
+      '',
+      {'database': 'password'},
+      loadYaml(serverpodConfig),
+    );
+
+    expect(config.database?.host, 'localhost');
+    expect(config.database?.port, 5432);
+    expect(config.database?.name, 'testDb');
+    expect(config.database?.user, 'test');
+    expect(config.database?.password, 'password');
+  });
+
+  test(
+      'Given Serverpod config with redis configuration missing required field when loading from Yaml then exception is thrown.',
+      () {
+    var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+redis:
+  port: 6379
+''';
+
+    expect(
+      () => ServerpodConfig.loadFromYaml(
+        '',
+        '',
+        {},
+        loadYaml(serverpodConfig),
+      ),
+      throwsA(isA<Exception>().having(
+        (e) => e.toString(),
+        'message',
+        equals('Exception: redis is missing required configuration for host.'),
+      )),
+    );
+  });
+
+  test(
+      'Given Serverpod config with redis configuration when loading from Yaml then config is loaded correctly.',
+      () {
+    var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+redis:
+  host: localhost
+  port: 6379
```

Comment:
Question: Redis also takes the key/value pair `enabled: bool` to toggle the config. Do we need a test for that here?

Reply:
The value doesn't toggle if the configuration is read or not. It is only a field on the configuration object with the default value false. Will treat it as other things with default values.